components/discord_webhook/actions/send-message-with-file/send-message-with-file.js

```diff
@@ -0,0 +1,73 @@
+const discordWebhook = require("../../discord_webhook.app.js");
+var request = require("request");
+const fs = require("fs");
+
+module.exports = {
+  key: "discord_webhook-send-message-with-file",
+  name: "Send Message With File",
+  description: "Post a message with an attached file",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    discordWebhook,
+    message: {
+      propDefinition: [
+        discordWebhook,
+        "message",
+      ],
+    },
+    threadID: {
+      propDefinition: [
+        discordWebhook,
+        "threadID",
+      ],
+    },
+    username: {
+      propDefinition: [
+        discordWebhook,
+        "username",
+      ],
+    },
+    avatarURL: {
+      propDefinition: [
+        discordWebhook,
+        "avatarURL",
+      ],
+    },
+    fileUrl: {
+      type: "string",
+      label: "File URL",
+      description:
+        "The URL of the file to attach. Must specify either File URL or File Path.",
+      optional: true,
+    },
+    filePath: {
+      type: "string",
+      label: "File Path",
+      description:
+        "The path to the file, e.g. /tmp/myFile.csv . Must specify either File URL or File Path.",
```

Comment:
Can you also add a runtime check for this to make this error clear to the user? We've found that users hit these cases a lot, so adding an error like this can help:

```javascript
if (!fileUrl && !filePath) {
  throw new Error("This action requires either File URL or File Path. Please enter one or the other above.");
}
```

@dannyroosevelt this use case doesn't come up a lot, but occasionally actions require _at least one of N_ props to work correct. For example, you can pass _either_ message text _or_ a "blocks" object (rich text messages) to the Slack Send a message action. Neither are required, but at least one of them is. We have no elegant way to surface that to the user, and I haven't checked to see what Zapier / Integromat do.

Reply:
Separately, can you update the description to read:

```
The path to the file, e.g. `/tmp/myFile.csv`. Must specify either **File URL** or **File Path**.
```

I'll add this to the component guidelines, but I'm norming on using `code` syntax anywhere there's a reference to a filepath / example JSON / other code, and **bold** when referencing the names of props.

Reply:
Added runtime check and updated descriptions